queries.sql

```diff
@@ -1,5 +1,111 @@
 /*Queries that provide answers to the questions from all projects.*/
 
+/* FEATURE BRANCH 04-add-join-table-for-visits */
+
+--D) Write queries to answer thhe following...
+
+-- D1) Who was the last animal seen by William Tatcher?
+SELECT animals.name as animal, vets.name as vet, 
+visits.date_of_visit
+FROM visits
+JOIN animals ON (visits.animal_id = animals.id) 
+JOIN vets ON (visits.vet_id = vets.id 
+	AND vets.name = 'William Tatcher')
+WHERE (visits.date_of_visit = (SELECT MAX(visits.date_of_visit) 
+	FROM visits 
+		JOIN animals ON (visits.animal_id = animals.id) 
+		JOIN vets ON (visits.vet_id = vets.id 
+			AND vets.name = 'William Tatcher')));
+
+-- D2) How many different animals did Stephanie Mendez see?
+SELECT animals.name as animal, vets.name as vet 
+FROM visits
+JOIN animals ON (visits.animal_id = animals.id) 
+JOIN vets ON (visits.vet_id = vets.id 
+	AND vets.name = 'Stephanie Mendez')
+GROUP BY animals.name, vets.name;
+
+-- D3) List all vets and their specialties, including vets with
+--     no specialties:
+SELECT v.name, s.species_id, sp.name as specialty
+FROM vets as v
+LEFT JOIN specializations as s ON (v.id = s.vet_id)
+LEFT JOIN species as sp ON (s.species_id = sp.id);
+
+-- D4) List all animals that visited Stephanie Mendez
+--     between April 1st and August 30th, 2020
+SELECT a.name as animal, ve.name as vet, 
+v.date_of_visit
+FROM visits as v
+JOIN animals as a ON (v.animal_id = a.id)
+JOIN vets as ve ON (v.vet_id = ve.id)
+ORDER BY (a.name, v.date_of_visit);
```

Comment:
- [ ] Hello! 🌟 Excellent job on this query, but it's crucial to narrow down the search for animals visited by Stephen Mendez to make it more specific. 🦸‍♂️ We're looking for animals visited by Stephen Mendez, but only during a specific timeframe, which is between April 1st and August 30th, 2020. 📅 So, let's focus on Stephen Mendez's visits during that period to create an accurate list of animals. 📆🐾